pkg/apis/serving/v1beta1/trained_model.go

```diff
@@ -0,0 +1,44 @@
+package v1beta1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:openapi-gen=true
+// +kubebuilder:object:root=true
+type TrainedModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              TrainedModelSpec   `json:"spec,omitempty"`
+	Status            TrainedModelStatus `json:"status,omitempty"`
+}
+
+// TrainedModelList contains a list of TrainedModel
+// +kubebuilder:object:root=true
+type TrainedModelList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// +listType=set
+	Items []TrainedModel `json:"items"`
+}
+
+type TrainedModelSpec struct {
+	// Required field for parent inference service
```

Comment:
If it's required, I think there's a +required annotation you can use instead.

Reply:
That's the CRD validation https://book.kubebuilder.io/reference/markers/crd-validation.html? Good to know! I will add them.

Reply:
I used `// +kubebuilder:validation:Required` to mark all the fields in this package as required and then labeled some optional fields using `// optional`